### PR TITLE
Improve remove filters link on search page

### PR DESCRIPTION
**Pull request by scruti**
[Jira ticket](https://regulatorydelivery.atlassian.net/browse/COSBETA-1309)

Instead of re-submitting the form clearing the filters in the submitted query, we're going to, using Javascript, reset the filters without submitting the form.

## Review apps

https://cosmetics-pr-2316-submit-web.london.cloudapps.digital/
https://cosmetics-pr-2316-search-web.london.cloudapps.digital/

### CODE REVIEW

File: cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb [hunk 1]
```diff
@@ -72,10 +72,25 @@
...
+      </button>
     </div>
-
   <% end %>
 </section>
+
+<%= javascript_tag nonce: true do -%>
+  # Clear filters reset
+  var filterForm = document.getElementById('new_notification_search_form_filters');
+  var ele1 = filterForm.querySelectorAll('.govuk-radios__conditional');
```
Comment (slawosz) [inline cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb:85]: I would try to rename variables.

File: cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb [hunk 2]
```diff
@@ -72,10 +72,25 @@
...
-      <% end %>
+      <button type="reset" class="govuk-button govuk-!-font-size-16 opss-button-link" id="opss-reset">
+        Reset
+      </button>
     </div>
-
   <% end %>
 </section>
+
+<%= javascript_tag nonce: true do -%>
```
Comment (slawosz) [inline cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb:82]: In general, its better to have entire JS in one place in `app/javascripts`, but at the same time, this one is not mission critical and probably can stay here. What do you think?

**Review by slawosz — APPROVED** [review]

File: cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb (hunk 2 above)
Comment (scruti) [reply cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb:82]: Going to move it to `notification_search.js`

**Comment by nasirkhanbeis** [comment]
Spotted few issues
1. When I apply filter and then reset isn't clearing the entered date or selected category from the dropdown
2. When I have a validation error on the date range field and then click reset, it isn't clearing the selected filters

File: cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb (hunk 1 above)
Comment (scruti) [reply cosmetics-web/app/views/poison_centres/notifications/_filters.html.erb]: Done!

**Review by slawosz — APPROVED** [review]

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looking all good as expected.

**Pull request merged**